meson.build

```diff
@@ -0,0 +1,123 @@
+project('libtexpdf', 'c', version : '2.4.6', default_options : ['c_std=gnu11'])
+
+pkg = import('pkgconfig')
+
+compiler = meson.get_compiler('c')
+
+conf = configuration_data()
+conf.set('version', meson.project_version())
+
+check_headers = ['dlfcn.h', 'inttypes.h', 'stdint.h', 'stdlib.h', 'string.h', 'sys/stat.h', 'sys/types.h', 'sys/wait.h', 'unistd.h']
+foreach header_name : check_headers
+    if compiler.has_header(header_name)
+        conf.set('HAVE_' + header_name.underscorify().to_upper(), 1)
+    endif
+endforeach
+
+if compiler.has_function('timezone', prefix : '#include <time.h>')
+    conf.set('HAVE_TIMEZONE', 1)
+endif
+
+if compiler.has_function('getenv', prefix : '#include <stdlib.h>')
+    conf.set('HAVE_GETENV', 1)
+endif
+
+if compiler.has_function('mkstemp', prefix : '#include <stdlib.h>')
+    conf.set('HAVE_MKSTEMP', 1)
+endif
+
+if compiler.has_member('struct tm', 'tm_gmtoff', prefix : '#include <time.h>')
+    conf.set('HAVE_TM_GMTOFF', 1)
+endif
+
+configure_file(input: 'config.h.meson', output: 'config.h', configuration: conf)
+
+libm = compiler.find_library('m', required : false)
+libdl = compiler.find_library('dl', required : false)
+
+zlib_dep = dependency('zlib')
+libpng_dep = dependency('libpng')
+
+libtexpdf_deps = [libpng_dep, zlib_dep, libdl, libm]
+
+libtexpdf_cargs = ['-DXETEX', '-DBUILDING_LIBTEXPDF', '-DHAVE_CONFIG_H', '-DHAVE_ZLIB', '-DHAVE_ZLIB_COMPRESS2', '-DHAVE_LIBPNG']
+
+libtexpdf_sources = [
+	  'agl.c',
+	  'bmpimage.c',
+	  'cff.c',
+	  'cff_dict.c',
+	  'cid.c',
+	  'cidtype0.c',
+	  'cidtype2.c',
+	  'cmap.c',
+	  'cmap_read.c',
+	  'cmap_write.c',
+	  'cs_type2.c',
+	  'dpxcrypt.c',
+	  'dpxfile.c',
+	  'dpxutil.c',
+	  'epdf.c',
+	  'error.c',
+	  'fontmap.c',
+	  'jp2image.c',
+	  'jpegimage.c',
+	  'mem.c',
+	  'mfileio.c',
+	  'numbers.c',
+	  'otl_conf.c',
+	  'otl_opt.c',
+	  'pdfcolor.c',
+	  'pdfdev.c',
+	  'pdfdoc.c',
+	  'pdfdraw.c',
+	  'pdfencrypt.c',
+	  'pdfencoding.c',
+	  'pdffont.c',
+	  'pdfnames.c',
+	  'pdfobj.c',
+	  'pdfparse.c',
+	  'pdfresource.c',
+	  'pdfximage.c',
+	  'pngimage.c',
+	  'pst.c',
+	  'pst_obj.c',
+	  'sfnt.c',
+	  'subfont.c',
+	  't1_char.c',
+	  't1_load.c',
+	  'truetype.c',
+	  'tt_aux.c',
+	  'tt_cmap.c',
+	  'tt_glyf.c',
+	  'tt_gsub.c',
+	  'tt_post.c',
+	  'tt_table.c',
+	  'type0.c',
+	  'type1.c',
+	  'type1c.c',
+	  'unicode.c',
+	  'pkfont.c',
+	  'tfm.c',
+]
+
+libtexpdf = library('texpdf', libtexpdf_sources,
```

Comment:
Note this is kind of weird but it is likely a good idea to do for compatibility reasons.

libtool by default sets the version to 0 if you don't explicitly specify one, and to suppress that you need to opt into `libtexpdf_la_LDFLAGS += -avoid-version`

Reply:
> Note this is kind of weird but it is likely a good idea to do for compatibility reasons.
> 
> libtool by default sets the version to 0 if you don't explicitly specify one, and to suppress that you need to opt into `libtexpdf_la_LDFLAGS += -avoid-version`. Having a soname version "0" doesn't really say anything meaningful, but at this point it's baked into the ABI so 🤷.

Thanks for bringing this up but I am confused. Either we declare the project version to zero which seems unorthodox and confusing or otherwise we use some obscure linker flag to set the version to zero. In the latter case I feel this can be confusing and difficult to support across all platforms. I don't even know how this can be done with Meson.

Reply:
In meson, you just set the `version: '0'` kwarg to the library.

This isn't related to the project version.

Reply:
> In meson, you just set the `version: '0'` kwarg to the library.

Ok, thank you, I didn't know about that option. That makes sense but before making this change I would prefer to have the advice of the other contributors. If it is up to me I would just omit this change.